azure_functions_worker/bindings/meta.py

```diff
@@ -93,6 +125,22 @@ def from_incoming_proto(
         raise TypeError(f'Unknown ParameterBindingType: {pb_type}')
 
     try:
+        # if the binding is an sdk type binding
+        if (SDK_BINDING_REGISTRY is not None
```

Comment:
nit: extract this logic to a helper function

Reply:
nit: deferred_bindings_enabled can be used here?

Reply:
Extracted logic into helper functions. Deferred_bindings_enabled can't be used here. We also need to check the binding type before calling the appropriate decode(). Moving this check to a single method in the base extension.